fda-model/paireddata/GraphicalUncertainPairedData.cs

```diff
@@ -62,15 +72,58 @@ public GraphicalUncertainPairedData(double[] exceedanceProbabilities, double[] f
             Graphical graphical = new Graphical(exceedanceProbabilities, flowOrStageValues, equivalentRecordLength, maximumProbability, minimumProbability, usingFlows, flowsAreNotLogged);
             graphical.ComputeGraphicalConfidenceLimits();
             _ExceedanceProbabilities = graphical.ExceedanceProbabilities;
-            _FlowOrStageDistributions = graphical.FlowOrStageDistributions;
+            _UnrevisedDistributions = graphical.FlowOrStageDistributions;
+            _DistributionsMonotonicFromAbove = MakeMeMonotonicFromAbove(_UnrevisedDistributions);
+            _DistributionsMonotonicFromBelow = MakeMeMonotonicFromBelow(_UnrevisedDistributions);
+            _EquivalentRecordLength = equivalentRecordLength;
             _metaData = new CurveMetaData(xlabel, ylabel, name, description);
         }
-
-
-
+        #endregion
 
         #region Functions
         IPairedData IPairedDataProducer.SamplePairedData(double probability)
+        {
+            double[] y = new double[_UnrevisedDistributions.Length];
+            if (probability > 0.5)
+            {
+                for (int i = 0; i < _ExceedanceProbabilities.Length; i++)
+                {
+                    y[i] = _DistributionsMonotonicFromAbove[i].InverseCDF(probability);
+                }
+            }
+            else
+            {
+                for (int i = 0; i < _ExceedanceProbabilities.Length; i++)
+                {
+                    y[i] = _DistributionsMonotonicFromBelow[i].InverseCDF(probability);
+                }
+            }
+            PairedData pairedData = new PairedData(_ExceedanceProbabilities, y, _metaData);
+            pairedData.Validate();
+            if (pairedData.HasErrors)
+            {
+                if (pairedData.RuleMap[nameof(pairedData.Yvals)].ErrorLevel > Base.Enumerations.ErrorLevel.Unassigned)
+                {
+                    Array.Sort(pairedData.Yvals);//sorts but doesnt solve the problem of repeated values.
```

Comment:
im not thrilled about this (i know you copied it from the uncertainpaireddata class, which i generally wrote...

i think we need to address this with something better than just randomly sorting it down the line.